components/core/src/Grep.cpp

```diff
@@ -131,12 +134,12 @@ QueryToken::QueryToken (const string& query_string, const size_t begin_pos, cons
 
             if (!converts_to_non_dict_var) {
                 // Dictionary variable
-                m_type = Type::DictOrIntVar;
+                m_type = Type::DictionaryVar;
                 m_cannot_convert_to_non_dict_var = true;
             } else {
                 m_type = Type::Ambiguous;
-                m_possible_types.push_back(Type::DictOrIntVar);
-                m_possible_types.push_back(Type::DoubleVar);
+                m_possible_types.push_back(Type::IntVar);
```

Comment:
Ideally, here still needs to consider Dict var
It's possible that the value has side wild card non-wildcard matches a dictionary value. 
For example, log: INFO static text ID: machine1234

where the query is "123"